### DISCUSSION TITLE
Question: is there a way to share files among different mount points?

### DISCUSSION THREAD

**Discussion by psantosl** [microsoft/ProjFS-Managed-API#47] [Q&A]
Suppose a mirror file system:

* Source is c:\src
* Mount in c:\mounted\copy00
* Another mount in c:\mounted\copy01

Is it possible to share the copies so if the file src\foo.c is not modified, the same file content is present in copy00 and copy01?

I mean, suppose foo.c is 10GB, what I want is to use just 10GB overall, not 10GB copied 3 times as the default behavior does.

(It would be some sort of copy on write).

**Comment by cgallred** [comment]
No, not entirely. It would be valid for your provider to create a hard link to c:\src\foo.c in its [PRJ_GET_PLACEHOLDER_INFO_CB](https://docs.microsoft.com/en-us/windows/win32/api/projectedfslib/nc-projectedfslib-prj_get_placeholder_info_cb) callback. But since there's no copy-on-write mechanism that would end up allowing the user to modify c:\src\foo.c if they had modify access to the file.
